builtin/providers/aws/resource_aws_autoscaling_group_waiting.go

```diff
@@ -0,0 +1,130 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// Waits for a minimum number of healthy instances to show up as healthy in the
+// ASG before continuing. Waits up to `waitForASGCapacityTimeout` for
+// "desired_capacity", or "min_size" if desired capacity is not specified.
+//
+// If "wait_for_elb_capacity" is specified, will also wait for that number of
+// instances to show up InService in all attached ELBs. See "Waiting for
+// Capacity" in docs for more discussion of the feature.
+func waitForASGCapacity(
+	d *schema.ResourceData,
+	meta interface{},
+	satisfiedFunc capacitySatisfiedFunc) error {
+	wait, err := time.ParseDuration(d.Get("wait_for_capacity_timeout").(string))
+	if err != nil {
+		return err
+	}
+
+	if wait == 0 {
+		log.Printf("[DEBUG] Capacity timeout set to 0, skipping capacity waiting.")
+		return nil
+	}
+
+	log.Printf("[DEBUG] Waiting on %s for capacity...", d.Id())
+
+	return resource.Retry(wait, func() error {
+		g, err := getAwsAutoscalingGroup(d, meta)
+		if err != nil {
+			return resource.RetryError{Err: err}
+		}
+		if g == nil {
+			return nil
+		}
+		lbis, err := getLBInstanceStates(g, meta)
+		if err != nil {
+			return resource.RetryError{Err: err}
+		}
+
+		haveASG := 0
+		haveELB := 0
+
+		for _, i := range g.Instances {
+			if i.HealthStatus == nil || i.InstanceId == nil || i.LifecycleState == nil {
+				continue
+			}
+
+			if !strings.EqualFold(*i.HealthStatus, "Healthy") {
+				continue
+			}
+
+			if !strings.EqualFold(*i.LifecycleState, "InService") {
+				continue
+			}
+
+			haveASG++
+
+			inAllLbs := true
+			for _, states := range lbis {
+				state, ok := states[*i.InstanceId]
+				if !ok || !strings.EqualFold(state, "InService") {
+					inAllLbs = false
+				}
+			}
+			if inAllLbs {
+				haveELB++
+			}
+		}
+
+		satisfied, reason := satisfiedFunc(d, haveASG, haveELB)
+
+		log.Printf("[DEBUG] %q Capacity: %d ASG, %d ELB, satisfied: %t, reason: %q",
+			d.Id(), haveASG, haveELB, satisfied, reason)
```

Comment:
This function is the same as the one removed from the resource file with the exception of these lines.